packages/docgen/lib/get-type-annotation.js

```diff
@@ -511,6 +511,33 @@ function getReturnTypeAnnotation( declarationToken ) {
 	}
 }
 
+/**
+ * @param {ASTNode} declarationToken
+ * @return {string} The type annotation for the variable.
+ */
+function getVariableTypeAnnotation( declarationToken ) {
+	let resolvedToken = declarationToken;
+	if ( babelTypes.isExportNamedDeclaration( resolvedToken ) ) {
+		resolvedToken = resolvedToken.declaration;
+	}
+
+	if ( babelTypes.isClassDeclaration( resolvedToken ) ) {
+		// just use the classname if we're exporting a class
+		return resolvedToken.id.name;
+	}
+
+	if ( babelTypes.isVariableDeclaration( resolvedToken ) ) {
+		resolvedToken = resolvedToken.declarations[ 0 ].id;
```

Comment:
What's the difference between this and the "exported named declaration"?

Reply:
An exported named declaration could be a `ClassDeclaration` or a `FunctionDeclaration` or a `VariableDeclaration`. `FunctionDeclaration`s are handled by the previous function type extraction work we've done whereas `ClassDeclaration` (handled above this) and `VariableDeclaration` (handled here) were excluded.